python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -109,13 +109,15 @@ async def get_chat_message_contents(
         # loop for auto-invoke function calls
         for request_index in range(settings.function_choice_behavior.maximum_auto_invoke_attempts):
             completions = await self._send_chat_request(settings)
-            # there is only one chat message, this was checked earlier
-            chat_history.add_message(message=completions[0])
-            # get the function call contents from the chat message
-            function_calls = [item for item in chat_history.messages[-1].items if isinstance(item, FunctionCallContent)]
+            # get the function call contents from the chat message, there is only one chat message
+            # this was checked earlier
+            function_calls = [item for item in completions[0].items if isinstance(item, FunctionCallContent)]
             if (fc_count := len(function_calls)) == 0:
                 return completions
 
+            # Since we have a function call, add the assistant's tool call message to the history
+            chat_history.add_message(message=completions[0])
+
```

Comment:
If we decided that this is the right fix, we need to fix it for all connectors.

Reply:
I'd prefer to keep this scoped to OpenAI, and then we can create a PR for the other connectors as needed.

Reply:
That sounds good to me! Should we also introduce another PR that fixes the issue in the `else` clause where function calling is not turned off?

Reply:
I decided to fix it for all connectors affected in this PR. Please have a look.